lib/internal/main/check_syntax.js

```diff
@@ -57,23 +57,23 @@ function loadESMIfNeeded(cb) {
 }
 
 async function checkSyntax(source, filename) {
-  let isModule = true;
+  let format;
   if (filename === '[stdin]' || filename === '[eval]') {
-    isModule = getOptionValue('--input-type') === 'module' ||
-      (getOptionValue('--experimental-default-type') === 'module' && getOptionValue('--input-type') !== 'commonjs');
+    format = (getOptionValue('--input-type') === 'module' ||
+      (getOptionValue('--experimental-default-type') === 'module' && getOptionValue('--input-type') !== 'commonjs')) ?
+      'module' : 'commonjs';
```

Comment:
This can probably be just

```suggestion
    format = getOptionValue('--input-type') || getOptionValue('--experimental-default-type');
```

Reply:
Those aren’t quite the same thing. I’m also nervous to rely on the fact that the potential values of both flags are exactly `commonjs` and `module`, and those are also the exact strings of the potential formats; if a future change causes a new value to either flag or a new format, this code would break. I can’t imagine that there’s a noticeable performance difference between the versions.